lib/internal/repl.js

```diff
@@ -84,18 +90,26 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
-    if (data) {
+    if (data && data[0] === '['
+        // Check the last two characters in case there is a trailing newline
+        && /\]/.test(data.substr(-2))) {
       try {
         repl.history = JSON.parse(data);
         if (!Array.isArray(repl.history)) {
           throw new Error('Expected array, got ' + typeof repl.history);
         }
         repl.history.slice(-repl.historySize);
```

Comment:
It shouldn't be permanently appending, since we're using [this variant](https://iojs.org/api/fs.html#fs_fs_write_fd_data_position_encoding_callback) of `fs.write`, with `position` set to `0`. The existing repl logic should be handling the trimming of history.